pubsub/google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py

```diff
@@ -392,17 +405,7 @@ def open(self, callback, on_callback_error):
         )
 
         # Create the RPC
-
-        # We must use a fixed value for the ACK deadline, as we cannot read it
-        # from the subscription. The latter would require `pubsub.subscriptions.get`
-        # permission, which is not granted to the default subscriber role
-        # `roles/pubsub.subscriber`.
-        # See also https://github.com/googleapis/google-cloud-python/issues/9339
-        #
-        # When dynamic lease management is enabled for the "on hold" messages,
-        # the default stream ACK deadline should again be set based on the
-        # historic ACK timing data, i.e. `self.ack_histogram.percentile(99)`.
-        stream_ack_deadline_seconds = _DEFAULT_STREAM_ACK_DEADLINE
+        stream_ack_deadline_seconds = self.ack_histogram.percentile(99)
```

Comment:
What is returned when the histogram has no data? I don't see a default value passed in when the histogram is created.

Reply:
It [returns 10](https://github.com/googleapis/google-cloud-python/blob/75dc3ecefb0ab093f812a81c0c2fccf42407ad10/pubsub/google/cloud/pubsub_v1/subscriber/_protocol/histogram.py#L152-L154) (seconds), which is the minimum ACK deadline supported by the backend. It essentially reverts [the change](https://github.com/googleapis/google-cloud-python/pull/9268/commits/20b717d637435b6765fbd7fee09dff26b2f0c5a4#diff-1004b3646a787f6b0438604df260553aR505) from one of the older commits.

(a possible separate cleanup task could be to get rid of this hardcoded value, and refactor Histogram to accept a default return value)